pkg/controllers/management/k3supgrade/register.go

```diff
@@ -37,6 +38,7 @@ func Register(ctx context.Context, wContext *wrangler.Context, mgmtCtx *config.M
 		systemUpgradeNamespace: systemUpgradeNS,
 		clusterCache:           wContext.Mgmt.Cluster().Cache(),
 		clusterClient:          wContext.Mgmt.Cluster(),
+		enqueue:                wContext.Mgmt.Cluster().Enqueue,
```

Comment:
Let's use engueueAfter just so we aren't stuck in a loop of checking even though we know the clusters is still ugprading.